edtLHD/src/main/java/fr/univtln/lhd/model/entities/dao/user/AdminDAO.java

```diff
@@ -26,18 +26,18 @@ public class AdminDAO implements DAO<Admin> {
 
     public AdminDAO() throws SQLException {
         Connection connection = Datasource.getInstance().getConnection();
-        this.get = connection.prepareStatement("SELECT * FROM MANAGERS WHERE ID=?");
-        this.getAll = connection.prepareStatement("SELECT * FROM MANAGERS");
-        this.save = connection.prepareStatement("INSERT INTO MANAGERS VALUES (DEFAULT, ?, ?, ?, ?, ?)",RETURN_GENERATED_KEYS);
-        this.update = connection.prepareStatement("UPDATE MANAGERS SET name=?, fname=? ,email=?,dpt=? WHERE ID=?");
-        this.delete = connection.prepareStatement("DELETE FROM MANAGERS WHERE ID=?");
+        this.get = connection.prepareStatement("SELECT * FROM ADMINS WHERE ID=?");
+        this.getAll = connection.prepareStatement("SELECT * FROM ADMINS");
+        this.save = connection.prepareStatement("INSERT INTO ADMINS VALUES (DEFAULT, ?, ?, ?, ?, ?)",RETURN_GENERATED_KEYS);
+        this.update = connection.prepareStatement("UPDATE ADMINS SET name=?, fname=? ,email=?,title=? WHERE ID=?");
+        this.delete = connection.prepareStatement("DELETE FROM ADMINS WHERE ID=?");
     }
```

Comment:
This  basically negates the whole idea of a connection pool, as the connection you will be using is gonna be on hold even when no operation is pending (its lifetime follows that of the class)

What you should instead do is 
1. make each of those static and or final strings 
2.  Make each of the method throw an Exception (whatever really)
3. in each method, use a try with ressources to get the conn. and the prepare statement :  the following works : 
```java 
//this.<whatever> := "Select from Subject where id = (?) "
//...
try( 
      Connection connection = Datasource.getConnection(); // I've added this in the Datasource in an earlier PR
       PreparedStatement stmt = connection.prepareStatement(this.<whatever>) // no ; required on the last one
       ) 
       {  /*resultSet.Set(...) or Get*/ }
catch (SqlException e) { throw e; /* or whatever */}

I see this is applicable to other DAOs as well, I can provide help if you so wish.

Reply:
hmmm, might do it later